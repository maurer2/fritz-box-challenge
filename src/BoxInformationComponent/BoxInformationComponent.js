import React, { PureComponent } from 'react';
import styled from 'styled-components/macro';
import PropTypes from 'prop-types';
import { CSSTransitionGroup } from 'react-transition-group';

import BoxEntryComponent from '../BoxEntryComponent/BoxEntryComponent';

const BoxInformationWrapper = styled.ul`
  display: flex;
  margin: 0;
  padding: 0;
  justify-content: space-between;
  flex-wrap: wrap;
  background: red;
`;

const BoxInformationEntry = styled.li`
  display: flex;
  flex-direction: column;
  flex-basis: 100%;
  flex-grow: 1;
  flex-shrink: 0;

  @media (min-width: 480px) {
    flex-basis: 0;
  }
`;

const SlideYTransition = styled.div`
  .slide-appear,
  .slide-enter {
    transform: translateY(100%);
  }

  .slide-appear.slide-appear-active,
  .slide-enter.slide-enter-active {
    transform: translateY(0);
    transition: transform 250ms ease-out;
  }

  .slide-leave {
    transform: translateY(0);
  }

  .slide-leave.slide-leave-active {
    transform: translateY(100%);
    transition: transform 250ms ease-out;
  }
`;

class BoxInformationComponent extends PureComponent {
  render() {
    const listFields = ['model', 'technology', 'restarts', 'branding'];

    return (
      <SlideYTransition>
        <CSSTransitionGroup
          component={ React.Fragment }
          transitionAppear={ false }
          transitionName="slide"
          transitionAppearTimeout={ 0 }
          transitionLeaveTimeout={ 250 }
          transitionEnterTimeout={ 250 }
        >
          { !this.props.isUpdating && (
            <BoxInformationWrapper>
              { listFields.map((entry, index) => (
                <BoxInformationEntry key={ index }>
                  <BoxEntryComponent entry={ entry } value={ this.props.list[entry] } />
                </BoxInformationEntry>
              ))}
            </BoxInformationWrapper>
          )}
        </CSSTransitionGroup>
      </SlideYTransition>
    );
  }
}

BoxInformationComponent.propTypes = {
  list: PropTypes.object,
  isUpdating: PropTypes.bool,
};

export default BoxInformationComponent;
