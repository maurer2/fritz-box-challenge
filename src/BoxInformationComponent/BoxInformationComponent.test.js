import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import BoxInformationComponent from './BoxInformationComponent';

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

describe('BoxInformationComponent', () => {
  const list = {
    branding: 'avm',
    model: 'FRITZ!Box 6590 Cable',
    restarts: '024',
    technology: 'Kabel',
  };

  const wrapper = shallow(<BoxInformationComponent list={ list } isUpdating={ false } />);

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
