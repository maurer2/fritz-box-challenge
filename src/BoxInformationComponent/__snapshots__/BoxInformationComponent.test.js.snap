// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxInformationComponent should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BoxInformationComponent
    isUpdating={false}
    list={
      Object {
        "branding": "avm",
        "model": "FRITZ!Box 6590 Cable",
        "restarts": "024",
        "technology": "Kabel",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <CSSTransitionGroup
        component={Symbol(react.fragment)}
        transitionAppear={false}
        transitionAppearTimeout={0}
        transitionEnter={true}
        transitionEnterTimeout={250}
        transitionLeave={true}
        transitionLeaveTimeout={250}
        transitionName="slide"
      >
        <ForwardRef>
          <ForwardRef>
            <BoxEntryComponent
              entry="model"
              value="FRITZ!Box 6590 Cable"
            />
          </ForwardRef>
          <ForwardRef>
            <BoxEntryComponent
              entry="technology"
              value="Kabel"
            />
          </ForwardRef>
          <ForwardRef>
            <BoxEntryComponent
              entry="restarts"
              value="024"
            />
          </ForwardRef>
          <ForwardRef>
            <BoxEntryComponent
              entry="branding"
              value="avm"
            />
          </ForwardRef>
        </ForwardRef>
      </CSSTransitionGroup>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ForwardRef>
          <ForwardRef>
            <BoxEntryComponent
              entry="model"
              value="FRITZ!Box 6590 Cable"
            />
          </ForwardRef>
          <ForwardRef>
            <BoxEntryComponent
              entry="technology"
              value="Kabel"
            />
          </ForwardRef>
          <ForwardRef>
            <BoxEntryComponent
              entry="restarts"
              value="024"
            />
          </ForwardRef>
          <ForwardRef>
            <BoxEntryComponent
              entry="branding"
              value="avm"
            />
          </ForwardRef>
        </ForwardRef>,
        "component": Symbol(react.fragment),
        "transitionAppear": false,
        "transitionAppearTimeout": 0,
        "transitionEnter": true,
        "transitionEnterTimeout": 250,
        "transitionLeave": true,
        "transitionLeaveTimeout": 250,
        "transitionName": "slide",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef>
              <BoxEntryComponent
                entry="model"
                value="FRITZ!Box 6590 Cable"
              />
            </ForwardRef>,
            <ForwardRef>
              <BoxEntryComponent
                entry="technology"
                value="Kabel"
              />
            </ForwardRef>,
            <ForwardRef>
              <BoxEntryComponent
                entry="restarts"
                value="024"
              />
            </ForwardRef>,
            <ForwardRef>
              <BoxEntryComponent
                entry="branding"
                value="avm"
              />
            </ForwardRef>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "children": <BoxEntryComponent
                entry="model"
                value="FRITZ!Box 6590 Cable"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "entry": "model",
                "value": "FRITZ!Box 6590 Cable",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "BoxInformationEntry-qkh4w1-1",
                "isStatic": true,
                "rules": Array [
                  "display:flex;flex-direction:column;flex-basis:100%;flex-grow:1;flex-shrink:0;@media (min-width:480px){flex-basis:0;}",
                ],
              },
              "displayName": "BoxInformationEntry",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "BoxInformationEntry-qkh4w1-1",
              "target": "li",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "children": <BoxEntryComponent
                entry="technology"
                value="Kabel"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "entry": "technology",
                "value": "Kabel",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "BoxInformationEntry-qkh4w1-1",
                "isStatic": true,
                "rules": Array [
                  "display:flex;flex-direction:column;flex-basis:100%;flex-grow:1;flex-shrink:0;@media (min-width:480px){flex-basis:0;}",
                ],
              },
              "displayName": "BoxInformationEntry",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "BoxInformationEntry-qkh4w1-1",
              "target": "li",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "children": <BoxEntryComponent
                entry="restarts"
                value="024"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "entry": "restarts",
                "value": "024",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "BoxInformationEntry-qkh4w1-1",
                "isStatic": true,
                "rules": Array [
                  "display:flex;flex-direction:column;flex-basis:100%;flex-grow:1;flex-shrink:0;@media (min-width:480px){flex-basis:0;}",
                ],
              },
              "displayName": "BoxInformationEntry",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "BoxInformationEntry-qkh4w1-1",
              "target": "li",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "function",
            "props": Object {
              "children": <BoxEntryComponent
                entry="branding"
                value="avm"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "entry": "branding",
                "value": "avm",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "BoxInformationEntry-qkh4w1-1",
                "isStatic": true,
                "rules": Array [
                  "display:flex;flex-direction:column;flex-basis:100%;flex-grow:1;flex-shrink:0;@media (min-width:480px){flex-basis:0;}",
                ],
              },
              "displayName": "BoxInformationEntry",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "BoxInformationEntry-qkh4w1-1",
              "target": "li",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "BoxInformationWrapper-qkh4w1-0",
            "isStatic": true,
            "rules": Array [
              "display:flex;margin:0;padding:0;justify-content:space-between;flex-wrap:wrap;background:red;",
            ],
          },
          "displayName": "BoxInformationWrapper",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "BoxInformationWrapper-qkh4w1-0",
          "target": "ul",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "SlideYTransition-qkh4w1-2",
        "isStatic": true,
        "rules": Array [
          "position:absolute;left:0;right:0;bottom:0;.slide-appear,.slide-enter{transform:translateY(100%);}.slide-appear.slide-appear-active,.slide-enter.slide-enter-active{transform:translateY(0);transition:transform 250ms ease-out;}.slide-leave{transform:translateY(0);}.slide-leave.slide-leave-active{transform:translateY(100%);transition:transform 250ms ease-out;}",
        ],
      },
      "displayName": "SlideYTransition",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "SlideYTransition-qkh4w1-2",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <CSSTransitionGroup
          component={Symbol(react.fragment)}
          transitionAppear={false}
          transitionAppearTimeout={0}
          transitionEnter={true}
          transitionEnterTimeout={250}
          transitionLeave={true}
          transitionLeaveTimeout={250}
          transitionName="slide"
        >
          <ForwardRef>
            <ForwardRef>
              <BoxEntryComponent
                entry="model"
                value="FRITZ!Box 6590 Cable"
              />
            </ForwardRef>
            <ForwardRef>
              <BoxEntryComponent
                entry="technology"
                value="Kabel"
              />
            </ForwardRef>
            <ForwardRef>
              <BoxEntryComponent
                entry="restarts"
                value="024"
              />
            </ForwardRef>
            <ForwardRef>
              <BoxEntryComponent
                entry="branding"
                value="avm"
              />
            </ForwardRef>
          </ForwardRef>
        </CSSTransitionGroup>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <ForwardRef>
            <ForwardRef>
              <BoxEntryComponent
                entry="model"
                value="FRITZ!Box 6590 Cable"
              />
            </ForwardRef>
            <ForwardRef>
              <BoxEntryComponent
                entry="technology"
                value="Kabel"
              />
            </ForwardRef>
            <ForwardRef>
              <BoxEntryComponent
                entry="restarts"
                value="024"
              />
            </ForwardRef>
            <ForwardRef>
              <BoxEntryComponent
                entry="branding"
                value="avm"
              />
            </ForwardRef>
          </ForwardRef>,
          "component": Symbol(react.fragment),
          "transitionAppear": false,
          "transitionAppearTimeout": 0,
          "transitionEnter": true,
          "transitionEnterTimeout": 250,
          "transitionLeave": true,
          "transitionLeaveTimeout": 250,
          "transitionName": "slide",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef>
                <BoxEntryComponent
                  entry="model"
                  value="FRITZ!Box 6590 Cable"
                />
              </ForwardRef>,
              <ForwardRef>
                <BoxEntryComponent
                  entry="technology"
                  value="Kabel"
                />
              </ForwardRef>,
              <ForwardRef>
                <BoxEntryComponent
                  entry="restarts"
                  value="024"
                />
              </ForwardRef>,
              <ForwardRef>
                <BoxEntryComponent
                  entry="branding"
                  value="avm"
                />
              </ForwardRef>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "children": <BoxEntryComponent
                  entry="model"
                  value="FRITZ!Box 6590 Cable"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "entry": "model",
                  "value": "FRITZ!Box 6590 Cable",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "BoxInformationEntry-qkh4w1-1",
                  "isStatic": true,
                  "rules": Array [
                    "display:flex;flex-direction:column;flex-basis:100%;flex-grow:1;flex-shrink:0;@media (min-width:480px){flex-basis:0;}",
                  ],
                },
                "displayName": "BoxInformationEntry",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "BoxInformationEntry-qkh4w1-1",
                "target": "li",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "children": <BoxEntryComponent
                  entry="technology"
                  value="Kabel"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "entry": "technology",
                  "value": "Kabel",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "BoxInformationEntry-qkh4w1-1",
                  "isStatic": true,
                  "rules": Array [
                    "display:flex;flex-direction:column;flex-basis:100%;flex-grow:1;flex-shrink:0;@media (min-width:480px){flex-basis:0;}",
                  ],
                },
                "displayName": "BoxInformationEntry",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "BoxInformationEntry-qkh4w1-1",
                "target": "li",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "function",
              "props": Object {
                "children": <BoxEntryComponent
                  entry="restarts"
                  value="024"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "entry": "restarts",
                  "value": "024",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "BoxInformationEntry-qkh4w1-1",
                  "isStatic": true,
                  "rules": Array [
                    "display:flex;flex-direction:column;flex-basis:100%;flex-grow:1;flex-shrink:0;@media (min-width:480px){flex-basis:0;}",
                  ],
                },
                "displayName": "BoxInformationEntry",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "BoxInformationEntry-qkh4w1-1",
                "target": "li",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "function",
              "props": Object {
                "children": <BoxEntryComponent
                  entry="branding"
                  value="avm"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "entry": "branding",
                  "value": "avm",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "BoxInformationEntry-qkh4w1-1",
                  "isStatic": true,
                  "rules": Array [
                    "display:flex;flex-direction:column;flex-basis:100%;flex-grow:1;flex-shrink:0;@media (min-width:480px){flex-basis:0;}",
                  ],
                },
                "displayName": "BoxInformationEntry",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "BoxInformationEntry-qkh4w1-1",
                "target": "li",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "BoxInformationWrapper-qkh4w1-0",
              "isStatic": true,
              "rules": Array [
                "display:flex;margin:0;padding:0;justify-content:space-between;flex-wrap:wrap;background:red;",
              ],
            },
            "displayName": "BoxInformationWrapper",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "BoxInformationWrapper-qkh4w1-0",
            "target": "ul",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "SlideYTransition-qkh4w1-2",
          "isStatic": true,
          "rules": Array [
            "position:absolute;left:0;right:0;bottom:0;.slide-appear,.slide-enter{transform:translateY(100%);}.slide-appear.slide-appear-active,.slide-enter.slide-enter-active{transform:translateY(0);transition:transform 250ms ease-out;}.slide-leave{transform:translateY(0);}.slide-leave.slide-leave-active{transform:translateY(100%);transition:transform 250ms ease-out;}",
          ],
        },
        "displayName": "SlideYTransition",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "SlideYTransition-qkh4w1-2",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
